### DISCUSSION TITLE
API missing a lot?

### DISCUSSION THREAD

**Discussion by mmuziek** [Zrips/CMI#5077] [Q&A]
Hello there im working on a bridge so that cmi icons will show up on bluemap.
however i have 2 questions that i would like to have some help with.

1. Apis are missing. for example i tried : ` jailmanager = CMI.getInstance().getJailManager();` but it returns a nullpointer.
2. same for the homes api i load up the getHomesManager and it returns a nullpointer? and no data.
3. if the homes api is working then i have another one. when i wanna loop the bedhomes i get a hashmap inside a hashmap and i wonder what the mapping is.
the first hashmap<string,hashmap> i have no idea what the string is. looking at the inner hashmap<cmiuser,cmihome> i know what both are but i have no clue if the hashmap is a single entry or multiple and i could not find the docs for it

**Comment by Zrips** [comment]
Updated API with 9.x file structure which should include latest methods.
